frontend/src/metabase/ui/utils/styled.ts

```diff
@@ -0,0 +1,21 @@
+import originalStyled from "@emotion/styled";
+
+type StylesParameters = Parameters<typeof originalStyled>;
+
+const shouldForwardProp = (prop: string) => !prop.startsWith("$");
```

Comment:
what if instead of renaming all these variables with `$`, we simply use emotion's built-in `isPropValid` ? like we do here?

https://github.com/metabase/metabase/blob/d819ee533db2e9b0f03cb7a54f0b91f27c4c8111/frontend/src/metabase/lib/styling/emotion.ts#L3-L5

https://emotion.sh/docs/styled#customizing-prop-forwarding

Reply:
ah, I see now, I missed that isPropValid helper somehow 👍